example/src/tests/groupTests.ts

```diff
@@ -1247,6 +1247,35 @@ test('can stream all groups and conversations', async () => {
   return true
 })
 
+test('can stream groups and messages', async () => {
+  for (const env of ['local', 'dev'] as const) {
+    const [alixClient, boClient] = await createClients(2, env)
+
+    // Start streaming groups
+    const groups: Group<any>[] = []
+    await alixClient.conversations.streamGroups(async (group: Group<any>) => {
+      groups.push(group)
+    })
+    // Also stream messages
+    await alixClient.conversations.streamAllMessages(
+      async (message) => {},
+      true
+    )
+
+    // bo creates a group with alix so a stream callback is fired
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    await boClient.conversations.newGroup([alixClient.address])
+    await delayToPropogate()
```

Comment:
I tested passing in a longer value to delayToPropogate here (default is 100 ms). Passing in 2000 ms, test passed 5/5 times for me on ios and android. There could just be network delays initiating the stream. 

To better deal with this flakiness, I think we need to update these tests to use Promise/Resolve instead of relying on delays, see this draft PR: https://github.com/xmtp/xmtp-react-native/pull/284/files

(I think at the time that PR was made, there was other issues that were breaking streams, but now that streams are more stable now might be a good time to revisit that improvement for tests)

Reply:
ah interesting, so I might have written a wrong test for a real problem I have in the app (that has nothing to do with 100ms I was waiting like 10 secs). I'll keep investigating then.